expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java

```diff
@@ -401,37 +402,27 @@ void shouldReturnExpandedResourceWithEntityDescriptionForPublicationWithContribu
     @Test
     void shouldReturnExpandedResourceWithCorrectNumberOfContributorsForPublicationWithSamePersonInDifferentRoles()
         throws JsonProcessingException, NotFoundException {
-        var publicationJsonString = stringFromResources(
-            Path.of("publication_sample.json"));
-
-        var publication = objectMapper.readValue(publicationJsonString, Publication.class);
-        var entityDescription = publication.getEntityDescription();
-        var contributors = entityDescription.getContributors();
         var id = randomUri();
-        var name = randomString();
-        var creator = createContributor(Role.CREATOR, id, name);
-        contributors.add(creator);
-        var actor = createContributor(Role.ACTOR, id, name);
-        contributors.add(actor);
-
+        var publication = getPublicationWithSamePersonInDifferentContributorRoles(id);
         var resourceUpdate = Resource.fromPublication(publication);
 
         expansionService = mockedExpansionService();
-
         var expandedResourceAsJson = expansionService.expandEntry(resourceUpdate).toJsonString();
 
         var regeneratedPublication = objectMapper.readValue(expandedResourceAsJson, Publication.class);
 
-        var contributorsWithSameId =
-            regeneratedPublication.getEntityDescription()
-                .getContributors()
-                .stream()
-                .filter(contributor -> contributor.getIdentity().getId().equals(id))
-                .collect(
-                    Collectors.toList());
+        var contributorsWithSameId = extractContributorsWithId(id, regeneratedPublication);
         assertThat(contributorsWithSameId.size(), is(equalTo(2)));
     }
 
+    private static List<Contributor> extractContributorsWithId(URI id, Publication publication) {
+        return publication.getEntityDescription()
+                   .getContributors()
+                   .stream()
+                   .filter(contributor -> contributor.getIdentity().getId().equals(id))
```

Comment:
Call to chained gets should be. extracted to a named method,  and the NPE problems should be managed :)